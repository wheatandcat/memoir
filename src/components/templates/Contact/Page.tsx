import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import * as Clipboard from 'expo-clipboard';
import { ConnectedType } from 'components/pages/Contact/Connected';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import { TextInput as RNTextInput } from 'react-native';
import useAutoFocusInput from 'hooks/useAutoFocusInput';

export type Props = ConnectedType & {};

const Page: React.FC<Props> = (props) => {
  const autoFocusProps = useAutoFocusInput(true, 500);

  const onCopyUserID = useCallback(() => {
    Clipboard.setString(props.userID);
  }, [props.userID]);

  const debug = false;

  return (
    <View style={styles.root}>
      {debug && (
        <>
          <View style={styles.title}>
            <Text color="secondaryLight" size="sm">
              ユーザーID
            </Text>
          </View>
          <View style={styles.userIDText}>
            <TouchableOpacity onPress={onCopyUserID}>
              <Text size="sm">{props.userID}</Text>
            </TouchableOpacity>
          </View>
        </>
      )}

      <View style={styles.title} pt={debug ? 4 : 0} pb={2}>
        <Text color="secondaryLight" size="sm">
          コメント
        </Text>
      </View>
      <RNTextInput
        multiline
        style={styles.inputText}
        autoCapitalize="none"
        onChangeText={props.onChangeText}
        maxLength={300}
        blurOnSubmit={false}
        {...autoFocusProps}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    paddingTop: theme().space(4),
  },
  title: {
    paddingVertical: theme().space(1),
    paddingHorizontal: theme().space(3),
  },
  userIDText: {
    backgroundColor: theme().color.background.light,
    paddingVertical: theme().space(3),
    paddingHorizontal: theme().space(3),
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: theme().color.secondary.light,
  },
  inputText: {
    borderColor: theme().color.secondary.light,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    paddingTop: theme().space(2),
    paddingBottom: theme().space(1),
    paddingHorizontal: theme().space(3),
    height: 250,
    color: theme().color.secondary.main,
    backgroundColor: theme().color.background.light,
    lineHeight: 20,
    fontWeight: 'bold',
    textAlignVertical: 'top',
  },
});

export default memo(Page);
