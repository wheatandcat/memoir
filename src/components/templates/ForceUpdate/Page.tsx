import React, { memo, useCallback } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import AppLink from 'react-native-app-link';
import Button from 'components/atoms/Button';

export type Props = {};
const appStoreId = 1613532672;
const playStoreId = 'com.wheatandcat.memoir';

const Page: React.FC<Props> = () => {
  const onPress = useCallback(() => {
    AppLink.openInStore({
      appName: 'memoir',
      appStoreId,
      appStoreLocale: 'jp',
      playStoreId,
    });
  }, []);

  return (
    <View style={styles.root}>
      <View py={5} mx={3}>
        <Text lineHeight={30}>
          新しいバージョンがリリースされました。{'\n'}
          ストアからアプリのバージョンアップをして下さい。
        </Text>
      </View>

      <Button title="ストアへ" width={250} onPress={onPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    padding: theme().space(3),
    height: '100%',
    alignItems: 'center',
  },
});

export default memo(Page);
