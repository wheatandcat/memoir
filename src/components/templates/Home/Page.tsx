import React, { memo, useState, useCallback } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import DateInput from 'components/organisms/DateInput/DateInput';
import MemoirButton from 'components/molecules/Home/MemoirButton.tsx';
import Cards from 'components/organisms/Cards/Cards';
import SettingModal from 'components/organisms/SettingModal';
import AddItemModal from 'components/organisms/AddItemModal';
import { LinearGradient } from 'expo-linear-gradient';
import theme from 'config/theme';

dayjs.locale('ja');
dayjs.extend(advancedFormat);

type Props = {
  date: string;
  openSettingModal: boolean;
  onItem: () => void;
  onMemoir: () => void;
  onCloseSettingModal: () => void;
  onChangeDate: (date: string) => void;
};

type State = {
  openAddItemModal: boolean;
};

const initialState = (): State => ({
  openAddItemModal: false,
});

const Page: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());

  const onAddItem = useCallback(() => {}, []);

  return (
    <LinearGradient
      // Background Linear Gradient
      colors={[theme().color.gradation[0], theme().color.gradation[1]]}
      style={styles.root}
    >
      <SettingModal
        isVisible={props.openSettingModal}
        onClose={props.onCloseSettingModal}
      />
      <AddItemModal
        isVisible={state.openAddItemModal}
        date={dayjs().format('YYYY-MM-DD')}
        onAdd={onAddItem}
        onClose={() => setState((s) => ({ ...s, openAddItemModal: false }))}
      />
      <ScrollView>
        <View style={styles.inner}>
          <DateInput date={props.date} onChange={props.onChangeDate} />
          <Cards
            onItem={props.onItem}
            onAddItem={() =>
              setState((s) => ({ ...s, openAddItemModal: true }))
            }
          />
        </View>
      </ScrollView>
      <MemoirButton onPress={props.onMemoir} />
    </LinearGradient>
  );
};

export default memo(Page);

const styles = StyleSheet.create({
  root: {
    height: '100%',
  },
  inner: {
    height: '100%',
  },
});
