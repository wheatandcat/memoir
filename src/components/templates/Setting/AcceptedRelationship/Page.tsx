import React, { memo } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Image from 'components/atoms/Image';
import Button from 'components/atoms/Button';
import theme from 'config/theme';
import { User } from 'store/atoms';
import UserImage from 'components/molecules/User/Image';

export type Props = {
  user: User;
  displayName: string;
  image: string;
};

const Page: React.FC<Props> = (props) => {
  const navigation = useNavigation();

  return (
    <View style={styles.root}>
      <ScrollView>
        <View style={styles.inner}>
          <View style={styles.imageWrap}>
            <View>
              <UserImage image={props.user.image || ''} size={100} />
            </View>
            <View mx={3}>
              <Text textAlign="center" size="xl2">
                ＋
              </Text>
            </View>
            <View>
              <UserImage image={props.image || ''} size={100} />
            </View>
          </View>
          <View m={4}>
            <Image
              source={require('../../../../img/icon/icon_cracker.png')}
              width={100}
              height={100}
            />
          </View>
          <View>
            <Text textAlign="center" size="lg">
              共有メンバーに{'\n'}なりました！
            </Text>
          </View>
          <View mt={5}>
            <Button
              title="一覧に戻る"
              width={200}
              onPress={() => {
                navigation.goBack();
              }}
            />
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inner: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  imageWrap: {
    paddingTop: theme().space(4),
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default memo(Page);
