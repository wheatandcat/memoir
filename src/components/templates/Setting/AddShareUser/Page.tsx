import React, { memo, useState, useEffect } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import { ConnectedType } from 'components/pages/Setting/AddShareUser/Connected';
import View from 'components/atoms/View';
import theme from 'config/theme';
import InviteCard from 'components/organisms/AddShareUser/InviteCard';
import InputInvite from 'components/organisms/AddShareUser/InputInvite';
import InputModal from 'components/organisms/AddShareUser/InputModal';
import { Invite } from 'components/pages/Setting/AddShareUser/Connected';
import FocusAwareStatusBar from 'components/organisms/FocusAwareStatusBar';
import usePrevious from 'hooks/usePrevious';

export type Props = ConnectedType & {
  invite: Invite;
};

const Page: React.FC<Props> = (props) => {
  const [dialog, setDialog] = useState<boolean>(false);
  const [isConfirm, setConfirm] = useState<boolean>(false);
  const confirmUserID = usePrevious(props.confirmUser?.id);
  const displayName = props.requestUser?.displayName ?? '';
  const [sentDisplayName, setSent] = useState<string>('');
  const preDisplayName = usePrevious(displayName);

  useEffect(() => {
    if (props.confirmUser?.id !== confirmUserID) {
      if (props.confirmUser?.id) {
        setConfirm(true);
      } else {
        setConfirm(false);
      }
    }
  }, [props.confirmUser?.id, confirmUserID]);

  useEffect(() => {
    if (displayName !== preDisplayName) {
      setSent(displayName);
    }
  }, [displayName, preDisplayName]);

  return (
    <>
      <FocusAwareStatusBar
        backgroundColor={theme().color.primary.main}
        style="light"
      />
      <InputModal
        isVisible={dialog}
        onClose={() => {
          setDialog(false);
          setConfirm(false);
          setSent('');
        }}
        onSearchInviteCode={props.onSearchInviteCode}
        displayName={sentDisplayName}
        requesting={props.requesting}
        isConfirm={isConfirm}
        confirmUser={props.confirmUser}
        onCreateRelationshipRequest={props.onCreateRelationshipRequest}
      />
      <ScrollView>
        <View style={styles.root}>
          <View style={styles.inner}>
            <InviteCard
              invite={props.invite}
              user={props.user}
              loading={props.loading}
              creating={props.creating}
              updating={props.updating}
              onCreateInvite={props.onCreateInvite}
              onUpdateInvite={props.onUpdateInvite}
            />
          </View>
          <View style={styles.inner}>
            <InputInvite onOpen={() => setDialog(true)} />
          </View>
        </View>
      </ScrollView>
    </>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
    marginTop: theme().space(3),
  },
  inner: {
    width: '90%',
    marginBottom: theme().space(4),
  },
});

export default memo(Page);
