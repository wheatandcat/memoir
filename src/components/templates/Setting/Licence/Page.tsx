import React, { memo } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import theme from 'config/theme';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import LicenceList from 'components/organisms/Setting/Licence/LicenceList';

type Props = {};

const Page: React.FC<Props> = () => {
  return (
    <View style={styles.root}>
      <ScrollView>
        <View style={styles.inner}>
          <View m={3}>
            <Text variants="small" color="secondaryLight">
              デザイン
            </Text>
          </View>
          <LicenceList />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
  },
  inner: {
    height: '100%',
  },
});

export default memo(Page);
