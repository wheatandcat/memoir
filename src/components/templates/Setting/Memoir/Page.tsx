import React, { memo, useState } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import SwitchSelector from 'react-native-switch-selector';
import { ConnectedType } from 'components/pages/Setting/Memoir/Connected';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import Input from 'components/organisms/Setting/Memoir/Input';

export type Props = ConnectedType & {};

const Page: React.FC<Props> = () => {
  const [dayOfWeek, setDayOfWeek] = useState(0);
  const [time, setTime] = useState(new Date('0000-01-01T00:00:00'));
  const [push, setPush] = useState(0);

  const options = [
    { label: 'OFF', value: 0 },
    { label: 'ON', value: 1 },
  ];

  return (
    <View style={styles.root}>
      <ScrollView>
        <View style={styles.inner}>
          <View>
            <View>
              <Text textAlign="center">いつ、ふりかえりますか？</Text>
            </View>
            <Input
              dayOfWeek={dayOfWeek}
              time={time}
              onChangeDayOfWeek={setDayOfWeek}
              onChangeTime={setTime}
            />
          </View>
          <View style={styles.title}>
            <View style={styles.line} />

            <View>
              <Text>通知設定</Text>
            </View>

            <View style={styles.line} />
          </View>
          <View style={styles.push}>
            <View>
              <Text variants="small">プッシュ通知</Text>
            </View>
            <View>
              <SwitchSelector
                initial={push}
                options={options}
                bold
                buttonColor={theme().color.primary.main}
                backgroundColor={theme().color.base.main}
                fontSize={theme().fontSizes.base}
                textColor={theme().color.base.light}
                selectedColor={theme().color.secondary.main}
                style={styles.pushInput}
                height={43}
                onPress={(v) => setPush(Number(v))}
              />
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
  inner: {
    paddingVertical: theme().space(3),
  },
  title: {
    marginTop: theme().space(5),
    flexDirection: 'row',
    height: 50,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  line: {
    width: '30%',
    backgroundColor: theme().color.base.dark,
    height: StyleSheet.hairlineWidth,
    marginHorizontal: theme().space(3),
  },
  pushInput: {
    width: 139,
  },
  push: {
    marginTop: theme().space(4),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
});

export default memo(Page);
