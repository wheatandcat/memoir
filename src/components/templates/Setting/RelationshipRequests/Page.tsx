import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import { ConnectedType } from 'components/pages/Setting/RelationshipRequests/Connected';
import View from 'components/atoms/View';
import theme from 'config/theme';
import { Props as PlainProps } from 'components/pages/Setting/RelationshipRequests/Plain';
import List from 'components/organisms/RelationshipRequest/List';

export type Props = ConnectedType &
  Pick<PlainProps, 'items' | 'loading' | 'onLoadMore' | 'pageInfo'>;

const Page: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <List {...props} />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    width: '100%',
  },
});

export default memo(Page);
