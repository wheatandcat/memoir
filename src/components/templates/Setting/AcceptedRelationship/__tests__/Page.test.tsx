import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { user } from '__mockData__/user';
import Page, { Props } from '../Page';

const propsData = (): Props => ({
  user: {
    ...user(),
    userID: '',
  },
  displayName: 'suzuki',
  image: '',
});

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigation: jest.fn(),
    }),
  };
});

describe('components/templates/Setting/AcceptedRelationship/Page.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Page {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
