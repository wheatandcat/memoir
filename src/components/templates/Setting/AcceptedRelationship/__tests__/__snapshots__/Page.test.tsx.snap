// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/templates/Setting/AcceptedRelationship/Page.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingTop": 32,
          }
        }
      >
        <View>
          <Memo(UserImage)
            image=""
            size={100}
          />
        </View>
        <View
          mx={3}
        >
          <Text
            fontFamily="RobotoCondensed-Bold"
            size="xl2"
            textAlign="center"
            variants="body"
          >
            ＋
          </Text>
        </View>
        <View>
          <Memo(UserImage)
            image=""
            size={100}
          />
        </View>
      </View>
      <View
        m={4}
      >
        <Image
          height={100}
          source={null}
          width={100}
        />
      </View>
      <View>
        <Text
          fontFamily="RobotoCondensed-Bold"
          size="lg"
          textAlign="center"
          variants="body"
        >
          共有メンバーに
          

          なりました！
        </Text>
      </View>
      <View
        mt={5}
      >
        <Button
          fontFamily="NotoSansJP-Bold"
          onPress={[Function]}
          size="base"
          title="一覧に戻る"
          width={200}
        />
      </View>
    </View>
  </ScrollView>
</View>
`;
